wm/stack.go

```diff
@@ -28,6 +34,29 @@ func (s *stack) removeFromStack(win desktop.Window) {
 		return
 	}
 	s.clients = append(s.clients[:pos], s.clients[pos+1:]...)
+
+	pos = -1
+	for i, w := range s.mappingOrder {
+		if w == win {
+			pos = i
+		}
+	}
+	if pos == -1 {
+		return
+	}
+	s.mappingOrder = append(s.mappingOrder[:pos], s.mappingOrder[pos+1:]...)
+}
+
+func (s *stack) getMappingOrder() []xproto.Window {
+	return s.getWindowsFromClients(s.mappingOrder)
+}
+
+func (s *stack) getWindowsFromClients(clients []desktop.Window) []xproto.Window {
```

Comment:
This doesn't seem to belong as a function on stack - it does not use any of it's state... perhaps it's another helper function within wm/desk or wm/client?

Reply:
Moved to Client.